src/azul/drs.py

```diff
@@ -269,21 +268,38 @@ def get_object(self,
         depends on the access method specified.
         """
         if access_id is None:
-            return self._get_object(object_id, access_method=access_method)
+            return self._get_object(drs_uri, access_method)
         else:
-            return self._get_object_access(object_id, access_id, access_method=access_method)
+            return self._get_object_access(drs_uri, access_id, access_method)
 
-    def _get_object(self, object_id: str, access_method: AccessMethod) -> str:
-        url = self.url / object_id
+    def _get_drs_url(self, drs_uri: str, access_id: Optional[str] = None) -> str:
+        """
+        Translate a DRS URI into a DRS URL. All query params included in the DRS
+        URI (eg '{drs_uri}?version=123') will be carried over to the DRS URL.
+        Only hostname-based DRS URIs (drs://<hostname>/<id>) are supported while
+        compact, identifier-based URIs (drs://[provider_code/]namespace:accession)
+        are not.
+        """
+        parsed = furl(drs_uri)
+        require(parsed.scheme == 'drs',
+                f'The DRS URI {drs_uri} does not have the correct scheme')
+        reject(':' in parsed.netloc or ':' in str(parsed.path),
```

Comment:
Why does a colon in netloc indicate the URI is not hostname based?

Same question for path.

Reply:
From: https://ga4gh.github.io/data-repository-service-schemas/preview/develop/docs/#_for_drs_clients
"The colon character is not allowed in a Hostname-based DRS URI, making it easy to tell them apart."

Hostname-based URI: `drs://<hostname>/<id>`
Compact Identifier-based URI: `drs://[provider_code/]namespace:accession`

The reason both `netloc` and `path` need to be checked for a colon is because the `provider_code` and following slash is optional in CI-based URI, which means when parsed by `furl` the colon could end up in the `netloc` or `path` depending if a `provider_code` was included or not.

Reply:
That's actually an issue with the DRS standard, making it hard to parse when following RFC3986.

https://tools.ietf.org/html/rfc3986#section-3

In `drs://foo:bar` (the variant without provider code), `foo:bar` matches the `hier-part` production of the RFC grammar.

In `drs://fubaz/foo:bar` (the variant with provider code), `foo:bar` matches the `path` production. Both are valid URIs but the fact that the `namespace` and `accession` URI components end up matching different production in the RFC grammar, depending on the case, is problematic. We may want to raise this with the author if the standard. A better alternative standard would be 

`drs://[provider_code]/namespace:accession`